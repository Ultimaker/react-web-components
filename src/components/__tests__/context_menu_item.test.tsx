// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import ContextMenuItem from '../context_menu_item';
import Button from '../button';


describe('The ContextMenuItem component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            label: 'Menu item',
            onClickHandler: jest.fn(),
            onCloseMenuHandler: jest.fn(),
        };
        wrapper = shallow(<ContextMenuItem {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onClickHandler).not.toHaveBeenCalled();
    });

    it('should call the click handler when item is clicked', () => {
        wrapper.find(Button).props().onClickHandler();
        expect(props.onClickHandler).toHaveBeenCalled();
        expect(props.onCloseMenuHandler).toHaveBeenCalled();
    });

    it('should render item as disabled', () => {
        wrapper.setProps({ disabled: true });
        expect(wrapper.find('.disabled')).toHaveLength(1);
    });
});
