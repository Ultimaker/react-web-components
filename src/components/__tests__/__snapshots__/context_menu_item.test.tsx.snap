// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContextMenuItem component should render 1`] = `
<DropDownMenuItem
  active={false}
  onClickHandler={[MockFunction]}
  onCloseMenuHandler={[MockFunction]}
/>
`;
